pdo_sqlsrv/core_results.cpp

```diff
@@ -335,13 +338,30 @@ sqlsrv_buffered_result_set::sqlsrv_buffered_result_set( sqlsrv_stmt* stmt TSRMLS
                 meta[i].length += sizeof( char ) + sizeof( SQLULEN ); // null terminator space
                 offset += meta[i].length;
                 break;
+            case SQL_CHAR:
+            case SQL_VARCHAR:
+                if ( meta[i].length == sqlsrv_buffered_result_set::meta_data::SIZE_UNKNOWN ) {
+                    offset += sizeof( void* );
+                }
+                else {
+                    // If encoding is set to UTF-8, the following types are not necessarily column size.
+                    // We need to call SQLGetData with c_type SQL_C_WCHAR and set the size accordingly. 
+                    if ( encoding == SQLSRV_ENCODING( CP_UTF8 )) {
```

Comment:
it is better to use SQLSRV_ENCODING_UTF8 syntax instead of SQLSRV_ENCODING( CP_UTF8 )